File: golang/mapedge.go
package gaia

import "fmt"
import "github.com/aporeto-inc/elemental"

// MapEdgeIdentity represents the Identity of the object
var MapEdgeIdentity = elemental.Identity{
	Name:     "mapedge",
	Category: "mapedges",
}

// MapEdgesList represents a list of MapEdges
type MapEdgesList []*MapEdge

// MapEdge represents the model of a mapedge
type MapEdge struct {
	// ID is the identifier of the object.
	ID string `json:"ID" cql:"-" bson:"-"`

	// AcceptedFlows tells how many accepted flows are represented by the edge
	AcceptedFlows int `json:"acceptedFlows" cql:"acceptedflows,omitempty" bson:"acceptedflows"`

	// ID of the destination resource
	DestinationID string `json:"destinationID" cql:"-" bson:"-"`

	// Name is the name of the entity
	Name string `json:"name" cql:"name,omitempty" bson:"name"`

	// RejectedFlows tells how many flows has been rejected.
	RejectedFlows int `json:"rejectedFlows" cql:"rejectedflows,omitempty" bson:"rejectedflows"`

	// ID of the source resource
	SourceID string `json:"sourceID" cql:"-" bson:"-"`
}

// NewMapEdge returns a new *MapEdge
func NewMapEdge() *MapEdge {

	return &MapEdge{}
}

// Identity returns the Identity of the object.
func (o *MapEdge) Identity() elemental.Identity {

	return MapEdgeIdentity
}

// Identifier returns the value of the object's unique identifier.
func (o *MapEdge) Identifier() string {

	return o.ID
}

func (o *MapEdge) String() string {

	return fmt.Sprintf("<%s:%s>", o.Identity().Name, o.Identifier())
}

// SetIdentifier sets the value of the object's unique identifier.
func (o *MapEdge) SetIdentifier(ID string) {

	o.ID = ID
}

// GetName returns the name of the receiver
func (o *MapEdge) GetName() string {
	return o.Name
}

// SetName set the given name of the receiver
func (o *MapEdge) SetName(name string) {
	o.Name = name
}

// Validate valides the current information stored into the structure.
func (o *MapEdge) Validate() error {

	errors := elemental.Errors{}

	if err := elemental.ValidateRequiredString("name", o.Name); err != nil {
		errors = append(errors, err)
	}

	if len(errors) > 0 {
		return errors
	}

	return nil
}

// SpecificationForAttribute returns the AttributeSpecification for the given attribute name key.
func (MapEdge) SpecificationForAttribute(name string) elemental.AttributeSpecification {

	return MapEdgeAttributesMap[name]
}

// AttributeSpecifications returns the full attribute specifications map.
func (MapEdge) AttributeSpecifications() map[string]elemental.AttributeSpecification {

	return MapEdgeAttributesMap
}

// MapEdgeAttributesMap represents the map of attribute for MapEdge.
var MapEdgeAttributesMap = map[string]elemental.AttributeSpecification{
	"ID": elemental.AttributeSpecification{
		AllowedChoices: []string{},
		Autogenerated:  true,
		Exposed:        true,
		Filterable:     true,
		Format:         "free",
		Identifier:     true,
		Name:           "ID",
		Orderable:      true,
		ReadOnly:       true,
		Type:           "string",
		Unique:         true,
	},
	"AcceptedFlows": elemental.AttributeSpecification{
		AllowedChoices: []string{},
		Autogenerated:  true,
		Exposed:        true,
		Name:           "acceptedFlows",
		Stored:         true,
		Type:           "integer",
	},
	"DestinationID": elemental.AttributeSpecification{
		AllowedChoices: []string{},
		Autogenerated:  true,
		Exposed:        true,
		Format:         "free",
		Name:           "destinationID",
		ReadOnly:       true,
		Type:           "string",
	},
	"Name": elemental.AttributeSpecification{
		AllowedChoices: []string{},
		Exposed:        true,
		Filterable:     true,
		Format:         "free",
		Getter:         true,
		Name:           "name",
		Orderable:      true,
		Required:       true,
		Setter:         true,
		Stored:         true,
		Type:           "string",
		Unique:         true,
	},
	"RejectedFlows": elemental.AttributeSpecification{
		AllowedChoices: []string{},
		Autogenerated:  true,
		Exposed:        true,
		Name:           "rejectedFlows",
		Stored:         true,
		Type:           "integer",
	},
	"SourceID": elemental.AttributeSpecification{
		AllowedChoices: []string{},
		Autogenerated:  true,
		Exposed:        true,
		Format:         "free",
		Name:           "sourceID",
		ReadOnly:       true,
		Type:           "string",
	},
}

File: python/gaia/mapedge.py
# -*- coding: utf-8 -*-

from pyelemental import RESTObject

class MapEdge(RESTObject):
    """ Represents a MapEdge in the 

        Notes:
            MapEdge describes a dependency between two resources.
    """

    def __init__(self, **kwargs):
        """ Initializes a MapEdge instance

          Notes:
              You can specify all parameters while calling this methods.
              A special argument named `data` will enable you to load the
              object from a Python dictionary

          Examples:
              >>> mapedge = MapEdge(id=u'xxxx-xxx-xxx-xxx', name=u'MapEdge')
              >>> mapedge = MapEdge(data=my_dict)
        """

        super(MapEdge, self).__init__()

        # Read/Write Attributes
        
        self._id = None
        self._acceptedflows = None
        self._destinationid = None
        self._name = None
        self._rejectedflows = None
        self._sourceid = None
        
        self.expose_attribute(local_name="ID", remote_name="ID")
        self.expose_attribute(local_name="acceptedFlows", remote_name="acceptedFlows")
        self.expose_attribute(local_name="destinationID", remote_name="destinationID")
        self.expose_attribute(local_name="name", remote_name="name")
        self.expose_attribute(local_name="rejectedFlows", remote_name="rejectedFlows")
        self.expose_attribute(local_name="sourceID", remote_name="sourceID")

        self._compute_args(**kwargs)

    def __str__(self):
        return '<%s:%s>' % (self.identity()["name"], self.identifier)

    def identifier(self):
        """ Identifier returns the value of the object's unique identifier.
        """
        
        return self.ID
        

    def setIdentifier(self, ID):
        """ SetIdentifier sets the value of the object's unique identifier.
        """
        
        self.ID = ID
        

    def identity(self):
        """ Identity returns the Identity of the object.
        """
        return mapedgeIdentity

    # Properties
    @property
    def ID(self):
        """ Get ID value.

          Notes:
              ID is the identifier of the object.

              
        """
        return self._id

    @ID.setter
    def ID(self, value):
        """ Set ID value.

          Notes:
              ID is the identifier of the object.

              
        """
        self._id = value
    
    @property
    def acceptedFlows(self):
        """ Get acceptedFlows value.

          Notes:
              AcceptedFlows tells how many accepted flows are represented by the edge

              
        """
        return self._acceptedflows

    @acceptedFlows.setter
    def acceptedFlows(self, value):
        """ Set acceptedFlows value.

          Notes:
              AcceptedFlows tells how many accepted flows are represented by the edge

              
        """
        self._acceptedflows = value
    
    @property
    def destinationID(self):
        """ Get destinationID value.

          Notes:
              ID of the destination resource

              
        """
        return self._destinationid

    @destinationID.setter
    def destinationID(self, value):
        """ Set destinationID value.

          Notes:
              ID of the destination resource

              
        """
        self._destinationid = value
    
    @property
    def name(self):
        """ Get name value.

          Notes:
              Name is the name of the entity

              
        """
        return self._name

    @name.setter
    def name(self, value):
        """ Set name value.

          Notes:
              Name is the name of the entity

              
        """
        self._name = value
    
    @property
    def rejectedFlows(self):
        """ Get rejectedFlows value.

          Notes:
              RejectedFlows tells how many flows has been rejected.

              
        """
        return self._rejectedflows

    @rejectedFlows.setter
    def rejectedFlows(self, value):
        """ Set rejectedFlows value.

          Notes:
              RejectedFlows tells how many flows has been rejected.

              
        """
        self._rejectedflows = value
    
    @property
    def sourceID(self):
        """ Get sourceID value.

          Notes:
              ID of the source resource 

              
        """
        return self._sourceid

    @sourceID.setter
    def sourceID(self, value):
        """ Set sourceID value.

          Notes:
              ID of the source resource 

              
        """
        self._sourceid = value
    

    # mapedgeIdentity represents the Identity of the object
mapedgeIdentity = {"name": "mapedge", "category": "mapedges", "constructor": MapEdge}